packages/mdc-textfield/foundation.js

```diff
@@ -141,13 +142,8 @@ class MDCTextFieldFoundation extends MDCFoundation {
 
     this.receivedUserInput_ = true;
 
-    const {target, type} = evt;
-    const {TEXT_FIELD_ICON} = MDCTextFieldFoundation.cssClasses;
-    const targetIsIcon = this.adapter_.eventTargetHasClass(target, TEXT_FIELD_ICON);
-    const eventTriggersNotification = type === 'click' || evt.key === 'Enter' || evt.keyCode === 13;
-
-    if (targetIsIcon && eventTriggersNotification) {
-      this.adapter_.notifyIconAction();
+    if (this.icon_) {
+      this.icon_.handleTextFieldInteraction(evt);
```

Comment:
This is still listening at the text field's root element which means the icon code still needs to filter it. Wasn't our thought the other day to actually listen to this event at the icon level? Then we might not even need the `eventTargetHasClass` icon adapter API...